# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1341423517 14400
#      Wed Jul 04 13:38:37 2012 -0400
# Node ID 355ff41f3f97daf526c1614edafbf94db123860d
# Parent  1abb7c083171b3900acbf226f8d8344ffeaa42e3
# Parent  e2ac6e2c588a3c33a3a53eea2f045d8b6cacd4c1
maint: periodic merge of stable to default

diff --git a/src/mex.cc b/src/mex.cc
--- a/src/mex.cc
+++ b/src/mex.cc
@@ -1509,17 +1509,17 @@ private:
 };
 
 // Matlab-style sparse arrays.
 
 class mxArray_sparse : public mxArray_matlab
 {
 public:
 
-  mxArray_sparse (mxClassID id_arg, int m, int n, int nzmax_arg,
+  mxArray_sparse (mxClassID id_arg, mwSize m, mwSize n, mwSize nzmax_arg,
                   mxComplexity flag = mxREAL)
     : mxArray_matlab (id_arg, m, n), nzmax (nzmax_arg),
       pr (calloc (nzmax, get_element_size ())),
       pi (flag == mxCOMPLEX ? calloc (nzmax, get_element_size ()) : 0),
       ir (static_cast<mwIndex *> (calloc (nzmax, sizeof (mwIndex)))),
       jc (static_cast<mwIndex *> (calloc (n + 1, sizeof (mwIndex))))
     { }
 
